### PR TITLE
Get rid of external library for abi generation

### PR DESCRIPTION
**Pull request by begmaroman**
# Description

Get rid of external library for abi generation

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

### CODE REVIEW

**Review by jelacamarko — APPROVED** [review]

File: consensus/polybft/contractsapi/bindings-gen/main.go [hunk 1]
```diff
@@ -447,7 +447,7 @@ import (
 		log.Fatal(err)
 	}
 
-	if err := ioutil.WriteFile("./consensus/polybft/contractsapi/contractsapi.go", output, 0600); err != nil {
+	if err = os.WriteFile("./consensus/polybft/contractsapi/contractsapi.go", output, 0600); err != nil {
```
Comment (Stefan-Ethernal) [inline consensus/polybft/contractsapi/bindings-gen/main.go]: WDYT about using `common.SaveFileSafe` function?

```suggestion
	if err = common.SaveFileSafe("./consensus/polybft/contractsapi/contractsapi.go", output, 0600); err != nil {
```

File: consensus/polybft/contractsapi/artifacts-gen/main.go [hunk 2]
```diff
@@ -191,22 +189,32 @@ func main() {
...
-	fl, err := os.Create(currentPath + "/../gen_sc_data.go")
+	output, err := format.Source([]byte(str))
 	if err != nil {
+		fmt.Println(str)
 		log.Fatal(err)
 	}
 
-	_, err = fmt.Fprintf(fl, "%#v", f)
-	if err != nil {
+	if err = os.WriteFile(currentPath+"/../gen_sc_data.go", output, 0600); err != nil {
```
Comment (Stefan-Ethernal) [inline consensus/polybft/contractsapi/artifacts-gen/main.go]: WDYT about using `common.SaveFileSafe` function?

```suggestion
	if err = common.SaveFileSafe(currentPath+"/../gen_sc_data.go", output, 0600); err != nil {
```

**Review by Stefan-Ethernal — APPROVED** [review]
Looks good. Leaving some small suggestions.

File: consensus/polybft/contractsapi/bindings-gen/main.go (hunk 1 above)
Comment (begmaroman) [reply consensus/polybft/contractsapi/bindings-gen/main.go]: Done

File: consensus/polybft/contractsapi/artifacts-gen/main.go (hunk 2 above)
Comment (begmaroman) [reply consensus/polybft/contractsapi/artifacts-gen/main.go]: Done

**Review by rachit77 — APPROVED** [review]

**Pull request merged**